tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/EnumSuite.scala

```diff
@@ -272,67 +272,120 @@ class EnumSuite extends BaseDottySuite {
-        tname("Some"),
-        Nil,
-        ctorp(List(tparam("x", "Int"))),
-        Nil
+    runTestAssert[Stat]("enum Option { @deprecated case Some(x: Int) }")(
+      enumWithCase(
+        "Option",
+        Defn.EnumCase(
+          List(Mod.Annot(Init(Type.Name("deprecated"), Name(""), Nil))),
+          tname("Some"),
+          Nil,
+          ctorp(List(tparam("x", "Int"))),
+          Nil
+        )
       )
     )
   }
 
   test("case-generic") {
     val generic = Type.Param(Nil, pname("X"), Nil, Type.Bounds(None, None), Nil, List(pname("Ord")))
-    runTestAssert[Stat]("case A[X: Ord]()")(
-      Defn.EnumCase(Nil, tname("A"), List(generic), ctorp(), Nil)
+    runTestAssert[Stat]("enum E { case A[X: Ord]() }")(
+      enumWithCase("E", Defn.EnumCase(Nil, tname("A"), List(generic), ctorp(), Nil))
     )
-    runTestAssert[Stat]("case A[X: Ord]")(
-      Defn.EnumCase(Nil, tname("A"), List(generic), ctor, Nil)
+    runTestAssert[Stat]("enum E { case A[X: Ord] }")(
+      enumWithCase("E", Defn.EnumCase(Nil, tname("A"), List(generic), ctor, Nil))
     )
   }
 
   test("case-extends") {
     val init = Init(Type.Apply(pname("Option"), List(pname("Nothing"))), anon, Nil)
-    runTestAssert[Stat]("case None extends Option[Nothing]")(
-      Defn.EnumCase(Nil, tname("None"), Nil, ctor, List(init))
+    runTestAssert[Stat]("enum Option { case None extends Option[Nothing] }")(
+      enumWithCase("Option", Defn.EnumCase(Nil, tname("None"), Nil, ctor, List(init)))
     )
   }
 
   test("case-extends-argument") {
     val init = Init(pname("Color"), anon, List(List(Lit.Int(65280))))
-    runTestAssert[Stat]("case Red extends Color(65280)")(
-      Defn.EnumCase(Nil, tname("Red"), Nil, ctor, List(init))
+    runTestAssert[Stat]("enum Color { case Red extends Color(65280) }")(
+      enumWithCase(
+        "Color",
+        Defn.EnumCase(
+          Nil,
+          Term.Name("Red"),
+          Nil,
+          Ctor.Primary(Nil, Name(""), Nil),
+          List(Init(Type.Name("Color"), Name(""), List(List(Lit.Int(65280)))))
+        )
+      )
     )
   }
 
   test("case-extends-argument-generic") {
     val init = Init(pname("Color"), anon, List(List(Lit.Int(65280))))
-    runTestAssert[Stat]("case Red[T](a: Int) extends Color(65280)")(
-      Defn
-        .EnumCase(Nil, tname("Red"), List(pparam("T")), ctorp(List(tparam("a", "Int"))), List(init))
+    runTestAssert[Stat]("enum Color { case Red[T](a: Int) extends Color(65280) }")(
+      enumWithCase(
+        "Color",
+        Defn.EnumCase(
+          Nil,
+          Term.Name("Red"),
+          List(Type.Param(Nil, Type.Name("T"), Nil, Type.Bounds(None, None), Nil, Nil)),
+          Ctor.Primary(
+            Nil,
+            Name(""),
+            List(List(Term.Param(Nil, Term.Name("a"), Some(Type.Name("Int")), None)))
+          ),
+          List(Init(Type.Name("Color"), Name(""), List(List(Lit.Int(65280)))))
+        )
+      )
+    )
+  }
+
+  test("enum-case-toplevel") {
```

Comment:
This test checks that we should fail to get an enum case in any other situation.